### PR TITLE
fix cors

### PR DESCRIPTION
**Pull request by felipeweb**
Related issue #181

### CODE REVIEW

File: middlewares/middlewares.go [hunk 1]
```diff
@@ -57,3 +59,14 @@ func JwtMiddleware(key string) negroni.Handler {
 	})
 	return negroni.HandlerFunc(jwtMiddleware.HandlerWithNext)
 }
+
+// Cors middleware
+func Cors(origin []string) negroni.Handler {
+	return negroni.HandlerFunc(func(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+		w.Header().Set("Access-Control-Allow-Origin", strings.Join(origin, ", "))
+		w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
```
Comment (marioidival) [inline middlewares/middlewares.go]: Add PATCH method here.

File: config/middlewares/middlewares_test.go [hunk 2]
```diff
@@ -181,3 +181,39 @@ func appTest() *negroni.Negroni {
 	n.UseHandler(r)
 	return n
 }
+
+func TestCors(t *testing.T) {
```
Comment (marioidival) [inline config/middlewares/middlewares_test.go]: Need test with allowed methods

File: middlewares/utils.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ func permissionByMethod(method string) (permission string) {
 func renderFormat(w http.ResponseWriter, recorder *httptest.ResponseRecorder, format string) {
 	for key, value := range recorder.Header() {
 		if len(value) > 0 {
```
Comment (avelino) [inline middlewares/utils.go]: Need this if?

Comment (felipeweb) [inline middlewares/utils.go]: no more i will remove thanks for catch

File: config/middlewares/middlewares_test.go [hunk 4]
```diff
@@ -181,3 +181,191 @@ func appTest() *negroni.Negroni {
...
+	defer server.Close()
+	resp, err := http.Post(server.URL, "application/json", nil)
+	if err != nil {
+		t.Fatal("expected run without errors but was", err)
+	}
+	if resp.Header.Get("Access-Control-Allow-Origin") != "*" {
+		t.Errorf("expected allow origin *, but got %q", resp.Header.Get("Access-Control-Allow-Origin"))
+	}
+	methods := resp.Header.Get("Access-Control-Allow-Methods")
+	if !strings.Contains(methods, "POST") {
```
Comment (cassiobotaro) [inline config/middlewares/middlewares_test.go]: por que não simplificar esses ifs com um for?

**Pull request merged**